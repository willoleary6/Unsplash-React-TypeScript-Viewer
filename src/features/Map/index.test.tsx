import React from "react";
import { Map } from ".";
import Enzyme, { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import configureStore from "redux-mock-store";
import thunk, { ThunkDispatch } from "redux-thunk";
import { AnyAction } from "redux";
import reducer, { GeoMapState, initialState } from "../../slices/GeoMap/GeoMapSlice";
import { Provider } from "react-redux";

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
interface mockStoreInterface {
    geoMap: GeoMapState;
}
const mockStore = configureStore<
    mockStoreInterface,
    ThunkDispatch<mockStoreInterface, unknown, AnyAction>
>(middlewares);

describe("MapWidget render testing", () => {
    let store = mockStore();
    beforeEach(() => {
        const mockState: mockStoreInterface = { geoMap: initialState };

        store = mockStore(mockState);
    });
    it("renders and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        // using mount here instead of shallow because we want to render the child components too
        const wrapper = mount(
            <React.StrictMode>
                <Provider store={store}>
                    <Map />
                </Provider>
            </React.StrictMode>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
