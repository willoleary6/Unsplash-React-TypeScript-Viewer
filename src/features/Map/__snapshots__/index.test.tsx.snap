// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapWidget render testing renders and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Map>
    <link
      crossOrigin=""
      href="https://unpkg.com/leaflet@1.7.1/dist/leaflet.css"
      integrity="sha512-xodZBNTC5n17Xt2atTPuE1HxjVMSvLVW9ocqUKLsCC5CXdbqCmblAshOMAS6/keqq/sMZMZ19scR4PsZChSR7A=="
      rel="stylesheet"
    />
    <script
      crossOrigin=""
      integrity="sha512-XQoYMqMTK8LvdxXYG3nZ448hOEQiglfqkJs1NOQV44cWnUrBc8PkAOcXy20w0vlaXaVUearIOBhiXZ5V3ynxwA=="
      src="https://unpkg.com/leaflet@1.7.1/dist/leaflet.js"
    />
    <div
      className="map-element-row"
    >
      <CoordinateInput
        searchForCoordinates={[Function]}
      >
        <div
          className="coordinate-input-container"
        >
          <div
            className="coordinate-input-controls"
          />
          <div
            className="coordinate-input-controls"
          >
            <input
              onChange={[Function]}
              onKeyUp={[Function]}
            />
            <button
              disabled={true}
              onClick={[Function]}
            >
              üîç
            </button>
          </div>
          <div
            className="coordinate-input-controls"
          />
          <p
            className="error-message"
          />
        </div>
      </CoordinateInput>
    </div>
    <div
      className="map-element-row"
    >
      <CoordinateDisplay
        coordinates={
          Object {
            "lat": 0,
            "lng": 0,
          }
        }
        displayClassName="latitude-value-container"
      >
        <div
          className="latitude-value-container"
        >
          <p
            className="longitude-latitude-titlebox"
          >
            Lat: 
            0
          </p>
          <p
            className="longitude-latitude-titlebox"
          >
            Lng: 
            0
          </p>
        </div>
      </CoordinateDisplay>
      <CoordinateDisplay
        coordinates={
          Object {
            "lat": 0,
            "lng": 0,
          }
        }
        displayClassName="longitude-value-container"
      >
        <div
          className="longitude-value-container"
        >
          <p
            className="longitude-latitude-titlebox"
          >
            Lat: 
            0
          </p>
          <p
            className="longitude-latitude-titlebox"
          >
            Lng: 
            0
          </p>
        </div>
      </CoordinateDisplay>
    </div>
    <div
      id="map"
    >
      <div
        id="crosshair"
      >
        (+)
      </div>
      <MapContainer
        center={
          Array [
            46.81914,
            9.62556,
          ]
        }
        className="leaflet-map-container"
        scrollWheelZoom={false}
        whenCreated={[Function]}
        zoom={15}
      >
        <div
          className="leaflet-map-container"
        >
          <MapEventHandlers />
          <ForwardRef(LeafComponent)
            attribution="¬© <a href=\\"https://www.openstreetmap.org/copyright\\">OpenStreetMap</a> contributors"
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />
          <RenderGeoJsonData />
          <ChangeMapView
            latitude={46.81914}
            longitude={9.62556}
          />
        </div>
      </MapContainer>
    </div>
    <div
      className="map-element-row"
    >
      <CoordinateDisplay
        coordinates={
          Object {
            "lat": 0,
            "lng": 0,
          }
        }
        displayClassName="latitude-value-container"
      >
        <div
          className="latitude-value-container"
        >
          <p
            className="longitude-latitude-titlebox"
          >
            Lat: 
            0
          </p>
          <p
            className="longitude-latitude-titlebox"
          >
            Lng: 
            0
          </p>
        </div>
      </CoordinateDisplay>
      <CoordinateDisplay
        coordinates={
          Object {
            "lat": 0,
            "lng": 0,
          }
        }
        displayClassName="longitude-value-container"
      >
        <div
          className="longitude-value-container"
        >
          <p
            className="longitude-latitude-titlebox"
          >
            Lat: 
            0
          </p>
          <p
            className="longitude-latitude-titlebox"
          >
            Lng: 
            0
          </p>
        </div>
      </CoordinateDisplay>
    </div>
  </Map>
</Provider>
`;
